GA Documentation and Release Notes Updates [WIP]

geekspertise opened the pull request:
This PR is the first draft of Service Mesh documentation and Release Notes for GA. It is a work in progress. 

This PR includes the following changes:

- New content for GA (OperatorHub, control plane templates)
- Removed notes about TP12 limitations (CNI, IOR, MT)
- Made MT the default installation story
- Started to update Supported Configurations module which now appears in Install docs and Release Notes
- Hid fixed issues in Release Notes
- Created "Service Mesh Architecture" folder for "Understanding service mesh"
- General cleanup and edits
- Added links in Next Steps

Can @tvieira @knrc @brian-avery and @rcernich and @JStickler start reviewing this WIP PR?

brian-avery commented on modules/ossm-vs-istio.adoc:
```diff
@@ -12,25 +12,29 @@ The current release of {ProductName} differs from the current upstream Istio com
+[id="ossm-mt-vs-clusterwide_{context}"]
+=== Differences between multi-tenant and cluster-wide installations
+
+The main difference between a multi-tenant installation and a cluster-wide installation is the scope of privileges used by the control plane deployments, for example, Galley and Pilot. The components no longer use cluster-scoped Role Based Access Control (RBAC) `ClusterRoleBinding`, but rely on project-scoped RBAC `RoleBinding`.
```
ClusterRbacConfig has been replaced with ServiceMeshRbacConfig and ServiceMeshPolicy replaces MeshPolicy.

brian-avery commented on modules/ossm-vs-istio.adoc:
```diff
@@ -12,25 +12,29 @@ The current release of {ProductName} differs from the current upstream Istio com
 [id="ossm-automatic-injection_{context}"]
 == Automatic injection
-The upstream Istio community installation automatically injects the sidecar to namespaces you have labeled.
+The upstream Istio community installation automatically injects the sidecar to projects you have labeled.
```
semantics, but upstream injects the sidecar to pods in the projects you have labeled.

brian-avery replied:
Same with below. Sidecar is a pod level concept, not a project level concept.

brian-avery commented on modules/ossm-vs-istio.adoc:
```diff
@@ -39,15 +43,14 @@ apiVersion: "rbac.istio.io/v1alpha1"
 metadata:
   name: httpbin-client-binding
-  namespace: httpbin
+  project: httpbin
```
This should still be namespace.

brian-avery replied:
Same for below yaml snippets. Yaml uses namespace and not project.

brian-avery commented on modules/ossm-vs-istio.adoc:
```diff
@@ -114,7 +112,7 @@ Manually created routes are not managed by the {ProductShortName}.
 [id="ossm-catch-all-domains_{context}"]
 === Catch-all domains
-{ProductName} does not support catch-all or wildcard domains. If {ProductShortName} finds a catch-all domain in the gateway definition, {ProductName} will create the route but link:https://docs.okd.io/latest/architecture/networking/routes.html#route-hostnames[relies on OpenShift to create a default hostname]. The route that {ProductShortName} creates will *not* be a catch-all route and will have  ...
+{ProductName} does not support catch-all or wildcard domains. If the {ProductShortName} finds a catch-all domain in the gateway definition, {ProductName} will create the route but link:https://docs.okd.io/latest/architecture/networking/routes.html#route-hostnames[relies on OpenShift to create a default hostname]. The route that {ProductShortName} creates will *not* be a catch-all route and will h ...
```
IOR is not part of GA.

brian-avery commented on modules/ossm-mt-projects.adoc:
```diff
@@ -0,0 +1,58 @@
+[id="ossm-mt-projects_{context}"]
+= Specify {ProductShortName} projects for multitenancy
+
+A multi-tenant control plane installation only affects projects configured as part of the {ProductShortName}. You must specify the projects associated with the {ProductShortName} in a `ServiceMeshMemberRoll` resource located in the same project as the `ServiceMeshControlPlane` resource and name it `default`.
```
Multi-tenant is the only option now.

dmitri-d commented on modules/ossm-control-plane-operatorhub-deploy.adoc:
```diff
@@ -0,0 +1,48 @@
+
+.Prerequisites
+
+* An account with cluster administrator access.
```
Cluster Admin privileges aren't needed here, only maistra-admin role (which grants permissions to create/update/delete controlplane CRs)

JStickler replied:
Where in the process is the maistra-admin role created?  We should document granting the role to users after it's created.

JStickler commented on service_mesh/service_mesh_install/installing-ossm.adoc:
```diff
@@ -4,7 +4,7 @@ include::modules/ossm-document-attributes.adoc[]
 toc::[]
 
-Installing the {ProductShortName} involves installing the Operator, and then creating and managing a custom resource definition file to deploy the control plane.
+Installing the {ProductShortName} involves installing the Operator, creating and managing a custom resource definition file to deploy the control plane, and creating a `ServiceMeshMemberRoll` to specify the namespaces associated with the {ProductShortName}.
```
Do we want to change "the Operator" to specify that (for production) we're actually installing four Operators?

JStickler commented on service_mesh/service_mesh_install/prepare-to-deploy-applications-ossm.adoc:
```diff
@@ -13,18 +13,14 @@ When you deploy an application into the {ProductShortName}, there are several di
 
-* Deploy applications on {ProductName}.
\ No newline at end of file
+* xref:../service_mesh_install/ossm-example-bookinfo.html#ossm-bookinfo-tutorial[Deploy Bookinfo] on {ProductName}.
```
I thought we weren't productizing Bookinfo?

brian-avery commented on modules/ossm-control-plane-deploy.adoc:
```diff
@@ -6,34 +6,11 @@
 .Procedure
 
-. Create a custom resource definition file named `istio-installation.yaml`.
+. Create a `ServiceMeshControlPlane` file named `istio-installation.yaml`.
```
There's a default in OLM. This step should also not be required if you use OLM to create it.

brian-avery commented on modules/ossm-installation-activities.adoc:
```diff
@@ -5,7 +5,7 @@
 = {ProductName} installation activities
 
-The {ProductName} installation process creates two different projects (namespaces):
+The {ProductName} installation process creates four different projects (also known as namespaces):
```
* istio-system is the control plane namespace. Changes depending on what the user decides to use. 

Under OLM all of the operators get installed in the same namespace.

JStickler replied:
Is that the "openshift-operators" namespace?

brian-avery replied:
It is, but I haven't set up an environment to test yet. Wanted to confirm.

JStickler replied:
The "openshift-operators" namespace is where they were installed when in my environment.

brian-avery commented on modules/ossm-master-configuration.adoc:
```diff
@@ -5,9 +5,11 @@
 = {ProductName}'s master configuration
 
-{ProductName} relies on a proxy sidecar within the application's pod to provide {ProductShortName} capabilities to the application. You can enable automatic sidecar injection or manage it manually. Red Hat recommends automatic injection using the annotation with no need to label namespaces. This ensures that your application contains the appropriate configuration for the {ProductShortName} upon d ...
+{ProductName} relies on a proxy sidecar within the application's pod to provide {ProductShortName} capabilities to the application. You can enable automatic sidecar injection or manage it manually. Red Hat recommends automatic injection using the annotation with no need to label projectss. This ensures that your application contains the appropriate configuration for the {ProductShortName} upon de ...
```
projects rather than projectss

brian-avery commented on modules/ossm-master-configuration.adoc:
```diff
@@ -5,9 +5,11 @@
 [id="ossm-master-configuration_{context}"]
 = {ProductName}'s master configuration
```
There's no longer any configuration in the master-configuration section. This is a remnant of OCP3.11 support in which the master node had to be configured to enable webhooks. This is enabled by default on OCP4. Should we move the contents somewhere else?

JStickler replied:
If we don't need it for 4.1, we should remove it.

brian-avery replied:
Meaning the headline should be discussing master configuration, but the content discusses how we handle sidecar injection.

JStickler commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,134 @@
+
+. Click *Subscribe*.
+
+. Navigate to *Catalog* -> *Operator Management*.
```
Confirmed, when you click Subscribe, the next page you see is the Operator Management screen.  The user doesn't have to navigate there.

luksa commented:
In modules/ossm-automatic-sidecar-injection.adoc there are two places that mention `true` as the value of the annotation, but the value should actually be `"true"` (it must be a string, not a boolean, otherwise you'll get an error).

luksa commented on modules/ossm-cr-istio-global.adoc:
```diff
@@ -56,12 +56,12 @@ See the OpenShift documentation on xref:../../scalability_and_performance/recomm
-|`0.12.0`
+|`1.0.0`
 
 |`hub`
```
I can't add a comment on the `imagePullSecret` row, but it should be `imagePullSecrets`

luksa replied:
Same for "cpu" below. It says "percentage of CPU", but should be "amount of CPU". Also not that users can also specify CPU resources in full cores (e.g. `2`), not just in millicores.

luksa commented on modules/ossm-cr-parameters.adoc:
```diff
@@ -5,9 +5,9 @@
 = Custom resource parameters
 
-The following examples illustrate use of the supported custom resource parameters for {ProductName} and the tables provide additional information about supported parameters.
+The following examples illustrate use of the `ServiceMeshControlPlane` parameters and the tables provide additional information about supported parameters.
```
I'd insert a table here explaining the top-level fields (which are then explained below): `istio`, `threescale`. And then another table showing the first level fields under `istio` - such as `global`, `gateways`, `mixer`, `pilot`, ...

geekspertise replied:
@brian-avery This would be a good idea for both community and product docs.

brian-avery replied:
Not fully understanding what's being suggested here. Are you suggesting something in line with https://maistra-1-0.maistra.io/docs/installation/installation-options/?

brian-avery replied:
Can we discuss this here: https://issues.jboss.org/browse/MAISTRA-881 and cover after GA?

luksa commented on modules/ossm-cr-pilot.adoc:
```diff
@@ -5,7 +5,7 @@
 [id="ossm-cr-pilot_{context}"]
 = Istio Pilot example
```
All these sections are called "example", but they also contain tables describing the parameters, so this should be called more like "Istio Pilot configuration".

geekspertise replied:
Fixed.

luksa commented on modules/ossm-cr-gateway.adoc:
```diff
@@ -5,13 +5,7 @@
+Here is an example that illustrates the Istio gateway parameters for the `ServiceMeshControlPlane` and a description of the available parameters with appropriate values.
 
 [source,yaml]
 ----
```
Example and the table below contain `ior_enabled`. Didn't we remove ior support?

luksa replied:
Also, indentation in the example looks inconsistent.

geekspertise replied:
The indentation is correct here: https://servicemesh--ocpdocs.netlify.com/openshift-enterprise/latest/service_mesh/service_mesh_install/customizing-installation-ossm.html#ossm-cr-gateway_customizing-installation-ossm

luksa commented on modules/ossm-cr-mixer.adoc:
```diff
@@ -5,7 +5,7 @@
+Here is an example that illustrates the Mixer parameters for the `ServiceMeshControlPlane` and a description of the available parameters with appropriate values.
 
 [source,yaml]
 ----
```
Most lines in this example are rendered in red because the indentation is wrong. `policy` and `telemetry` should be at the same level as `enabled`.

geekspertise replied:
Fixed.

luksa commented on modules/ossm-cr-pilot.adoc:
```diff
@@ -5,7 +5,7 @@
+Here is an example that illustrates the Istio Pilot parameters for the `ServiceMeshControlPlane` and a description of the available parameters with appropriate values.
 
 [source,yaml]
 ----
```
Indentation in the `autoscaleEnabled` and `traceSampling` lines is off.

geekspertise replied:
@luksa How so? They're both indented at the same level, which is identical to the example on the Maistra site.

brian-avery replied:
GitHub link to the file from here isn't working for some reason, but the preview looks right to me.  
pilot:
    resources:
      requests:
        cpu: 100m
     autoscaleEnabled: false
     traceSampling: 100.0

brian-avery replied:
Actually, I think I see it. 

```
pilot:
   resources:
     requests:
       cpu: 100m
    autoscaleEnabled: false
    traceSampling: 100.0
```
Pilot to resources is two spaces. Resources to requests is two spaces. Pilot to autoscaleEnabled and traceSampling looks like it's three spaces.

geekspertise replied:
Fixed.

luksa commented on modules/ossm-cr-tracing-jaeger.adoc:
```diff
@@ -16,7 +16,6 @@ This example illustrates tracing parameters for the {ProductName} custom resourc
       jaeger:
```
Indentation is off. Should be two spaces throughout.

geekspertise replied:
@luksa This entire page was rewritten and is now under ossm-configuring-jaeger.adoc.

luksa commented on modules/ossm-member-roll.adoc:
```diff
@@ -0,0 +1,92 @@
+// * service_mesh/service_mesh_install/installing-ossm.adoc
+
+[id="ossm-member-roll_{context}"]
+= Configure the {ProductName} member roll
```
Member Roll should probably be capitalized, because the product name is also capitalized.

geekspertise replied:
The product name has to be capitalized, but we use sentence case headings at Red Hat not title case headings.

luksa commented on modules/ossm-vs-istio.adoc:
```diff
@@ -10,27 +10,31 @@ An installation of {ProductName} differs from upstream Istio community installat
 
 [id="ossm-multi-tenant-install_{context}"]
-== Multi-tenant installations
+== {ProductName} control plane
```
Control Plane should probably be capitalized because the product name is also capitalized.

geekspertise replied:
The product name is capitalized, but we use sentence case titles at Red Hat.

luksa commented on modules/ossm-member-roll.adoc:
```diff
@@ -0,0 +1,92 @@
+[id="ossm-member-roll_{context}"]
+= Configure the {ProductName} member roll
+
+You must specify the projects associated with the {ProductShortName} in a `ServiceMeshMemberRoll` resource located in the same project as the `ServiceMeshControlPlane` resource and name it `default`.
```
Strange sentence - starts off with "must specify", but finished with "and name it", which would make more sense if the sentence started with "must create a SMMR resource named default in order to specify the projects associated..."

geekspertise replied:
I changed this to read:

"You must create a `ServiceMeshMemberRoll` resource named `default` associated with the {ProductShortName} in the same project as the `ServiceMeshControlPlane`."

luksa commented on modules/ossm-member-roll.adoc:
```diff
@@ -0,0 +1,92 @@
+
+[WARNING]
+====
+If Container Network Interface (CNI) is enabled, manual sidecar injection will work, but pods will not be able to communicate with the control plane unless those pods are a part of the `ServiceMeshMemberRoll` resource.
```
This should be "If the Istio CNI (Container Network Interface) _plugin_ is enabled, ...".

Also: instead of "are a part of the SMMR resource", it might be clearer to say "are specified/listed in the SMMR resource".

geekspertise replied:
Fixed.

luksa commented on modules/ossm-member-roll.adoc:
```diff
@@ -0,0 +1,92 @@
+
+* You can add any number of projects, but a project can only belong to *one* `ServiceMeshMemberRoll`.
+
+* `ServiceMeshMemberRoll` resources are reconciled in response to the following events:
```
Not sure if this section is needed. As a user, I expect the SMMR to be reconciled whenever it needs to be, just like any other resource. I defined the desired state in the SMMR and I expect the operator to _always_ try to bring the actual state to how I want it to be. Thus, going into detail of when the operator reconciles the SMMR seems unnecessary. Also, when introducing the SMCP resource, we never explain when it gets reconciled.

geekspertise replied:
Removed.

luksa commented on modules/ossm-member-roll.adoc:
```diff
@@ -0,0 +1,92 @@
+[id="ossm-member-roll-console_{context}"]
+== Configure the member roll from the the {product-title} web console
+
+This example joins the Bookinfo project to the {ProductShortName} from the web console.
```
This ends up rendering as "... joins the Bookinfo project to the Service Mesh from the web console". Given how there can be multiple service meshes, the sentence is somewhat misleading as it makes it seem as though there's only one mesh. Plus, the product name is used here - if it changes, this sentence may no longer make sense.

geekspertise replied:
I've changed this to read:
"Follow this procedure to add the Bookinfo project to the {ProductShortName} member roll by using the web console."

luksa commented on modules/ossm-member-roll.adoc:
```diff
@@ -0,0 +1,92 @@
+
+. Click the *Project* menu and choose `openshift-operators` from the list.
+
+. Click the *Istio Service Mesh Member Roll* link under *Provided APIS* for the *{ProductName}* Operator.
```
Should this be APIs instead of APIS?

geekspertise replied:
Fixed.

luksa commented on modules/ossm-member-roll.adoc:
```diff
@@ -0,0 +1,92 @@
+.Prerequisites
+
+* An installed, verified {ProductShortName} Operator.
+* Location of the project where the custom resource installed the control plane.
```
Unclear what "Location" means. Should this be "Name" instead?

Also: might not be clear what "custom resource" refers to. Should be "ServiceMeshControlPlane resource". (I'd leave out the word "custom", as it only makes sense in the context of Kubernetes/OpenShift; in the context of the service mesh, it doesn't).

geekspertise replied:
Updated to read: 

"Name of the project with the `ServiceMeshMemberRoll` resource."

luksa commented on modules/ossm-member-roll.adoc:
```diff
@@ -0,0 +1,92 @@
+
+.Procedure
+
+. Create a custom resource file named `ServiceMeshMemberRoll` in the same project as the `ServiceMeshControlPlane` custom resource.
```
Multiple problems in this sentence:
- it's not a file, it's an object/resource
- ServiceMeshMemberRoll is not the name, but the kind/type of the object
- SMCP custom resource should just be "SMCP resource"

geekspertise replied:
Changed to read:

"Create `ServiceMeshMemberRoll` object in the same project as the `ServiceMeshControlPlane` resource."

luksa commented on modules/ossm-member-roll.adoc:
```diff
@@ -0,0 +1,92 @@
+
+. Name the resource `default`.
+
+. Add the projects to the member list in the `ServiceMeshMemberRoll`. The `bookinfo` project is joined to the {ProductShortName} in this example.
```
Not clear which projects. Should just be "the bookinfo project" or "the projects you wish to include in the service mesh".

Also: "is joined to the {ProductShortName}..." should be "is joined to the service mesh deployed in the same project as the SMMR resource"

geekspertise replied:
Updated this to read:

"Add the Bookinfo project to the member list in the `ServiceMeshMemberRoll`. In this example, the `bookinfo` project is joined to the {ProductShortName} deployed in the same project as the `ServiceMeshMembereRoll` resource."

luksa commented on modules/ossm-control-plane-deploy.adoc:
```diff
@@ -2,38 +2,60 @@
+. Click the {ProductName} Operator.
+
+. Under *Provided APIs*, the Operator enables you to create two resource types:
+** A {ProductName} control plane
```
This should be "A `ServiceMeshControlPlane` resource"

geekspertise replied:
Fixed.

luksa commented on service_mesh/service_mesh_install/installing-ossm.adoc:
```diff
@@ -46,10 +42,10 @@ include::modules/ossm-cr-threescale.adoc[leveloffset=+2]
+include::modules/ossm-member-roll.adoc[leveloffset=+1]
 
-include::modules/ossm-control-plane-verify.adoc[leveloffset=+2]
+include::modules/ossm-control-plane-deploy.adoc[leveloffset=+1]
```
This section should come before the ossm-cr sections above.

geekspertise replied:
This is the assembly file that pulls together the modules. Please see the preview attached to this PR to see how the modules are arranged.

luksa commented on modules/ossm-rn-new-features.adoc:
```diff
@@ -11,6 +11,19 @@ Reason – If known, include why has the enhancement been implemented (use case,
+
+Other notable changes in this release include the following:
+
+* The Kubernetes Container Network Interface (CNI) is always on.
```
This needs to say "Istio CNI plugin", as CNI is always present (and "enabled") in every Kubernetes cluster. What the service mesh actually installs is just a CNI _plugin_ that the CNI support inside Kubernetes calls out to when starting up a pod.

geekspertise replied:
Fixed.

jwendell commented:
The bookinfo section is outdated. See the most recent content here: https://maistra-1-0.maistra.io/docs/examples/bookinfo/

objectiser commented on modules/ossm-cr-tracing-jaeger.adoc:
```diff
@@ -42,10 +41,5 @@ This example illustrates tracing parameters for the {ProductName} custom resourc
 | The deployment template to use for Jaeger
 | The name of a template type
 | `all-in-one`/`production-elasticsearch`
```
"all-in-one" and "production-elasticsearch".

Sorry for the delay, didn't see the notification.

objectiser commented on modules/ossm-cr-tracing-jaeger.adoc:
```diff
@@ -31,7 +30,7 @@ This example illustrates tracing parameters for the {ProductName} custom resourc
 | The hub that the Operator uses to pull Jaeger images
 | A valid image repo
-| `jaegertracing/` or `registry.redhat.io/openshift-istio-tech-preview/`
+| `jaegertracing/` or `registry.redhat.io/openshift-service-mesh/`
```
its not `registry.redhat.io/openshift-service-mesh` but `registry.redhat.io/distributed-tracing`.

Don't think the values should have the `/` at the end, as the template includes that.

geekspertise replied:
This entire file was rewritten and is now in the ossm-configuring-jaeger.adoc file.

rcernich commented on modules/ossm-supported-configurations.adoc:
```diff
@@ -18,4 +19,13 @@ OpenShift Online and OpenShift Dedicated are not supported for {ProductName} {Pr
 * This release of {ProductName} is only available on {product-title} x86_64.
 * {ProductName} is only suited for {product-title} Software Defined Networking (SDN) configured as a flat network with no external providers.
```
I'm not sure what we officially support, but I believe we should be able to support all SDN modes: NetworkPolicy, Multitenant, and Subnet (flat network).

geekspertise replied:
@knrc Can you clarify this for GA?

geekspertise replied:
Given the short amount of time left, this can be addressed post-GA.

rcernich commented on service_mesh/service_mesh_install/preparing-ossm-installation.adoc:
```diff
@@ -9,17 +9,20 @@ Before you can install {ProductName}, review the installation activities, ensure
 
 * Possess an active {product-title} subscription on your Red Hat account. If you do not have a subscription, contact your sales representative for more information.
+* Review the xref:../../architecture/architecture-installation.adoc#installation-overview_architecture-installation[{product-title} 4.1 overview].
 * Install {product-title} 4.1.
```
Do we need to add details for special configuration required by Elasticsearch (used by Jaeger)?  My understanding is that the default machine configuration for nodes won't support deploying a pod that requires 16Gi of memory.  I don't know if those details belong here or with the Jaeger portion of the documentation.

brian-avery replied:
I think this is being covered in the Jaeger docs

objectiser replied:
@JStickler Do you know where this will be documented?

geekspertise replied:
The updated Jaeger Configuration is in the ossm-configuring-jaeger.adoc file.

rcernich commented on modules/ossm-rn-new-features.adoc:
```diff
@@ -11,6 +11,19 @@ Reason – If known, include why has the enhancement been implemented (use case,
+Other notable changes in this release include the following:
+
+* The Kubernetes Container Network Interface (CNI) is always on.
+* The control plane is configured for multitenancy by default. Single tenant, cluster-wide control plane configurations are deprecated.
```
Single tenant, cluster-wide is no longer supported.

geekspertise replied:
That's why I clarified that they are deprecated.

rcernich commented on modules/ossm-installation-activities.adoc:
```diff
@@ -5,12 +5,11 @@
 = {ProductName} installation activities
 
-The {ProductName} installation process creates two different projects (namespaces):
+The {ProductName} installation process creates the `openshift-operators` project (also known as a namespace) with pods for:
```
I think this entire section can be deleted.  The wording in the `Installing service mesh` section is well written and describes what needs to be done to install service mesh.  I think the only thing we really need to say is what operators need to be installed and/or the operators involved and what each governs (istio, elasticsearch, jaeger, kiali).  Maybe a blurb about working with operator hub, but like I said, the installation doc covers all of that.

geekspertise replied:
I removed the top of this section but kept the list of Operators and added Elasticsearch. Please see the next update to my PR.

rcernich commented on modules/ossm-cr-example.adoc:
```diff
@@ -7,10 +7,10 @@
 ====
 
-To deploy the {ProductShortName} control plane, you must create a custom resource. A _custom resource_ allows you to introduce your own API into a Kubernetes project or cluster. You create a custom resource yaml file that defines the project parameters and creates the object. This example custom resource yaml file contains all of the supported parameters and deploys {ProductName} {ProductVersion} ...
+To deploy the {ProductShortName} control plane, you must create a custom resource. A _custom resource_ allows you to extend the API in a Kubernetes project or cluster. You create a custom resource yaml file that defines the project parameters and creates the object. This example custom resource yaml file contains all of the supported parameters and deploys {ProductName} {ProductVersion} images ba ...
```
I like this text.

geekspertise replied:
I modified the Marko's text a bit and updated this section.

rcernich commented on modules/ossm-tutorial-bookinfo-verify-install.adoc:
```diff
@@ -10,17 +10,10 @@ Before configuring your application, verify that it successfully deployed.
+* Run this command to confirm that Bookinfo is deployed:
 +
 ----
 $ curl -o /dev/null -s -w "%{http_code}\n" http://<GATEWAY_URL>/productpage
```
Should this be something simpler like:

Run this command to confirm that Bookinfo is deployed:

`$ oc get pods -n bookinfo`

Verify that all pods are ready.

brian-avery replied:
The problem with doing this is you're only making sure the pod is running. Not that it's actually running, attached to the mesh, and routable via the Istio ingress gateway.

geekspertise replied:
Updated this with the @travisrogers05 changes.

rcernich commented on modules/ossm-tutorial-bookinfo-removing.adoc:
```diff
@@ -10,8 +10,9 @@ When you finish with the Bookinfo application, you can remove it by running the
+
 .Procedure
 
 . Download the cleanup script:
```
The link below is dead.  It should be sufficient to replicate the `apply` commands, using `delete` instead.  For example,

```
$ oc delete -n bookinfo -f https://raw.githubusercontent.com/Maistra/bookinfo/maistra-1.0/bookinfo-gateway.yaml
$ oc delete -n bookinfo -f -f https://raw.githubusercontent.com/Maistra/bookinfo/maistra-1.0/bookinfo.yaml
```

You'll also need to add a command to delete the destination rules, depending on which file they installed (i.e. whether using tls or not).

geekspertise replied:
The install script process is deprecated. Please see the updated file.

rcernich commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,90 @@
+// * service_mesh/service_mesh_install/removing-ossm.adoc
+
+[id="ossm-operatorhub-remove_{context}"]
+= Removing the Operators from OperatorHub
```
Maybe, `Removing the Operators Installed from OperatorHub`

geekspertise replied:
I changed this to read:

"Removing the installed Operators"

rcernich commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,90 @@
+. Click the *kiali-ossm* Options menu {kebab}, and then click *Remove Subscription*.
+
+. Select *Also completely remove the Operator from the selected namespace* and click *Remove*.
+
```
Should we add a blurb identifying additional steps required to fully purge the install?  Some resources are created by the operator that are not removed through the above procedure.  See https://issues.jboss.org/browse/MAISTRA-846.

This text should look something like:

The operator creates a number of resources, which are not managed by OLM (OperatorHub) and need to be removed manually after uninstalling the Istio operator.  As cluster-admin execute the following commands (where <operator-namespace> is the namespace within which the Istio operator was installed, typically `openshift-operators`):
```
$    oc delete validatingwebhookconfiguration/<operator-namespace>.servicemesh-resources.maistra.io
$    oc delete -n <operator-namespace> daemonset/istio-node
$    oc delete clusterrole/istio-admin
$    oc get crds -o name | grep '.*\.istio\.io' |xargs -r -n 1 oc delete
$    oc get crds -o name | grep '.*\.maistra\.io' |xargs -r -n 1 oc delete
```

geekspertise replied:
I added a procedure for this and it will appear in my next PR update.

brian-avery commented on modules/ossm-control-plane-templates.adoc:
```diff
@@ -0,0 +1,78 @@
+[id="ossm-control-plane-templates_{context}"]
+= Creating control plane templates
+
+When you configure control plane templates, which follow the same syntax as the `ServiceMeshControlPlane`, users inherit settings in a hierarchical factor. The Operator is delivered with a `default` template with default settings for {ProductName}. To add custom templates you must create a ConfigMap named `smcp-templates` in the `openshift-operators` and mount the ConfigMap in the Operator at `/u ...
```
Control planes are a new concept here as far as I can tell. Given that, is there a better way to start than "when you configure control plane templates"? 

This seems to be a couple of ideas mixed together.
* You can take a servicemesh control plane that you've created and tested and turn it into a template by including it in the `smcp-templates` configmap. These templates can build off of one another inheriting (accounting team -> default template for Service Mesh -> common settings between Maistra and Service Mesh)
* The operator ships with a temlate named default. If an organization chooses to, they can override the default template in the configmap, which means that all templates will build off of this.

rcernich replied:
Sounds like there should be a section describing what templates are and how/why they are used, followed up by a blurb describing how to create a custom template, followed by the blurb on how to configure the operator to use the custom template.

geekspertise replied:
This module has been changed significantly. Please review the updates.

brian-avery commented on service_mesh/service_mesh_arch/understanding-ossm.adoc:
```diff
@@ -6,8 +6,6 @@ toc::[]
 
 {ProductName} is a platform that provides behavioral insight and operational control over the service mesh. It gives you a uniform way to connect, secure, and monitor microservices in your {product-title} environment.
```
I like it.

geekspertise replied:
I'm fine with it and it's going to appear in my next PR update.

vikram-redhat commented:
@openshift/team-documentation Please be aware of this and the following two PRs coming in for peer review. 

https://github.com/openshift/openshift-docs/pull/16314
https://github.com/openshift/openshift-docs/pull/16317

The GA for SM is 4/Sep, so a quick turn around is required. I am pointing this out now since Monday is a holiday and we are at a cross-over point for the peer review roster.

@geekspertise @JStickler - any reason the team can't start the peer review now?

geekspertise commented:
> @openshift/team-documentation Please be aware of this and the following two PRs coming in for peer review.
> 
> #16314
> #16317
> 
> The GA for SM is 4/Sep, so a quick turn around is required. I am pointing this out now since Monday is a holiday and we are at a cross-over point for the peer review roster.
> 
> @geekspertise @JStickler - any reason the team can't start the peer review now?

@vikram-redhat The cutoff for feedback was COB yesterday. I have not addressed an enormous amount of feedback. The OpenShift team won't be able to start their review until Tuesday at the earliest.

hhovsepy commented:
Verified the release notes from Kiali side.

geekspertise commented on modules/ossm-control-plane-deploy.adoc:
```diff
@@ -2,38 +2,60 @@
+
+. Click *Create* to create the control plane.
+
+. The Operator starts up the pods, services, and other `ServiceMeshControlPlane` components.
```
I changed this to read:
"Service Mesh control plane components."

geekspertise commented on modules/ossm-control-plane-deploy.adoc:
```diff
@@ -2,38 +2,60 @@
+
+= Deploying the control plane from the CLI
+
+After you create your `ServiceMeshControlPlane` resource, deploy the {ProductShortName} control plane.
```
Changed this introduction to read: 
"Follow this procedure to deploy the Red Hat OpenShift Service Mesh control plane from the CLI.

geekspertise commented on modules/ossm-control-plane-remove.adoc:
```diff
@@ -0,0 +1,64 @@
+
+[id="ossm-control-plane-remove_{context}"]
+= Removing the {ProductName} control plane
+You can remove the {ProductShortName} control plane from OperatorHub or from the command line.
```
Changed this to read:
"You can remove the {ProductShortName} control plane by using the {product-title} web console or the command line."

geekspertise commented on modules/ossm-control-plane-remove.adoc:
```diff
@@ -0,0 +1,64 @@
+[id="ossm-control-plane-remove-operatorhub_{context}"]
+== Removing the control plane using OperatorHub
+
+Follow this procedure to remove the {ProductName} control plane from OperatorHub.
```
Replaced "from" with "by using".

geekspertise commented on modules/ossm-control-plane-remove.adoc:
```diff
@@ -0,0 +1,64 @@
+
+
+[id="ossm-control-plane-remove-cli_{context}"]
+== Removing the control plane from the CLI
```
Replaced "from" with "by using".

geekspertise commented on modules/ossm-control-plane-templates.adoc:
```diff
@@ -0,0 +1,78 @@
+[id="ossm-control-plane-templates_{context}"]
+= Creating control plane templates
+
+When you configure control plane templates, which follow the same syntax as the `ServiceMeshControlPlane`, users inherit settings in a hierarchical factor. The Operator is delivered with a `default` template with default settings for {ProductName}. To add custom templates you must create a ConfigMap named `smcp-templates` in the `openshift-operators` and mount the ConfigMap in the Operator at `/u ...
```
Corrected spelling of fashion and "project" is the preferred language in OpenShift.

geekspertise commented on modules/ossm-control-plane-templates.adoc:
```diff
@@ -0,0 +1,78 @@
+
+. Log into your {product-title} installation as a cluster administrator.
+
+. Run this command to create the ConfigMap and substitute `templates-directory` with the location of the `ServiceMeshControlPlane` files:
```
Fixed.

geekspertise commented on modules/ossm-control-plane-templates.adoc:
```diff
@@ -0,0 +1,78 @@
+       - name: istio-operator
+         spec:
+         ...
+                   volumeMounts:
```
@brian-avery Can you help me out? This affects community docs as well.

geekspertise replied:
I've updated this to match what @brian-avery put in the community docs.

geekspertise commented on modules/ossm-cr-example.adoc:
```diff
@@ -7,10 +7,10 @@
 
```
Fixed.

geekspertise commented on modules/ossm-cr-istio-global.adoc:
```diff
@@ -16,8 +16,8 @@ In order for the 3scale Istio Adapter to work, `disablePolicyChecks` must be `fa
     global:
-      hub: `maistra/` or `registry.redhat.io/openshift-istio-tech-preview/`
-      tag: 0.12.0
+      hub: `maistra/` or `registry.redhat.io/openshift-service-mesh/`
```
I removed them and let's see if there are any objections.

geekspertise commented on modules/ossm-cr-parameters.adoc:
```diff
@@ -5,9 +5,9 @@
 [id="ossm-cr-parameters_{context}"]
 = Custom resource parameters
```
Fixed.

geekspertise commented on modules/ossm-installation-activities.adoc:
```diff
@@ -5,12 +5,11 @@
-* observability project (1 pod)
+* {ProductShortName}
+* Kiali
+* Observability (Jaeger)
```
I added Elasticsearch to the list of Operators with a definition below.

geekspertise commented on modules/ossm-tutorial-bookinfo-removing.adoc:
```diff
@@ -6,37 +6,49 @@ This TASK module included in the following assemblies:
-No resources found.
-$ oc get pods -n myproject
-No resources found.
+$ oc -n <control plane project> patch --type='json' smmr default -p '[{"op": "remove", "path": "/spec/members", "value":["'"${bookinfo}"'"]}]'
```
Fixed.

geekspertise commented on modules/ossm-installation-activities.adoc:
```diff
@@ -5,12 +5,11 @@
+* Kiali
+* Observability (Jaeger)
 
 You first create a Kubernetes _Operator_. This Operator defines and monitors a _custom resource_ that manages the deployment, updating, and deletion of the {ProductShortName} components.
```
See the update to this section in my next PR update.

geekspertise commented on modules/ossm-control-plane-templates.adoc:
```diff
@@ -0,0 +1,78 @@
+maistra.v1.0.0            Red Hat OpenShift Service Mesh   1.0.0                Succeeded
+----
+
+. Edit the Operator cluster service version to instruct the Operator to use the `scmp-templates` ConfigMap.
```
Fixed.

geekspertise commented on modules/ossm-installation-activities.adoc:
```diff
@@ -5,14 +5,13 @@
-You first create a Kubernetes _Operator_. This Operator defines and monitors a _custom resource_ that manages the deployment, updating, and deletion of the {ProductShortName} components.
+You first deploy the {ProductName} Operator. This Operator defines and monitors a _custom resource_ that manages the deployment, updating, and deletion of the {ProductShortName} components.
 
 Depending on how you define the custom resource file, you can install one or more of the following components when you install the {ProductShortName}:
```
I rearranged this section and it should be more clear now.

geekspertise commented on modules/ossm-cr-example.adoc:
```diff
@@ -25,6 +27,7 @@ The 3scale Istio Adapter is deployed and configured in the custom resource file.
   kind: ServiceMeshControlPlane
   metadata:
     name: basic-install
```
@travisrogers05 Feel free to provide a full-install example and I will add it to the docs.

brian-avery replied:
If we're going to link to one, I'd suggest this one which is public:

https://github.com/Maistra/istio-operator/blob/maistra-1.0/deploy/examples/maistra_v1_servicemeshcontrolplane_cr_full.yaml

geekspertise replied:
I've added the full example that @brian-avery shared. If this needs to be modified, we can do that in a post-GA update.

jkandasa commented on modules/ossm-configuring-jaeger.adoc:
```diff
@@ -0,0 +1,131 @@
+        enabled: true
+        ingress:
+          enabled: true
+
```
Template files: 
* [jaeger-all-in-one.yaml](https://github.com/Maistra/istio-operator/blob/maistra-1.0/helm/istio/charts/tracing/templates/jaeger-all-in-one.yaml)
* [jaeger-production-elasticsearch.yaml](https://github.com/Maistra/istio-operator/blob/maistra-1.0/helm/istio/charts/tracing/templates/jaeger-production-elasticsearch.yaml)

geekspertise replied:
@JStickler can address this post-GA.

luksa commented on service_mesh/service_mesh_install/installing-ossm.adoc:
```diff
@@ -4,52 +4,38 @@ include::modules/ossm-document-attributes.adoc[]
 toc::[]
 
-Installing the {ProductShortName} involves installing the Operator, and then creating and managing a custom resource definition file to deploy the control plane.
+Installing the {ProductShortName} involves installing the Elasticsearch, Jaeger, Kiali and {ProductShortName} Operators, creating and managing a custom resource definition file to deploy the control plane, and creating a `ServiceMeshMemberRoll` to specify the namespaces associated with the {ProductShortName}.
```
Replace "...a custom resource definition file to deploy..." with "...a `ServiceMeshControlPlane` resource to deploy...`.

Also: add "resource" after `ServiceMeshMemberRoll`.

geekspertise replied:
Fixed.

luksa commented on modules/ossm-vs-istio.adoc:
```diff
@@ -10,27 +10,44 @@ An installation of {ProductName} differs from upstream Istio community installat
+[id="ossm-mt-vs-clusterwide_{context}"]
+=== Multi-tenancy in {ProductName} versus cluster-wide installations
+
+The main difference between a multi-tenant installation and a cluster-wide installation is the scope of privileges used by the control plane deployments, for example, Galley and Pilot. The components no longer use cluster-scoped Role Based Access Control (RBAC) `ClusterRoleBinding`, but rely on {ProductShortName} project-scoped RBAC `ServiceMeshRbacConfig`.
```
Change last sentence to:
"The components no longer use cluster-scoped Role Based Access Control (RBAC) resource `ClusterRoleBinding`, but rely on project-scoped `RoleBinding`."

ServiceMeshRbacConfig is something completely different. :)

geekspertise replied:
Fixed.

luksa commented on modules/ossm-vs-istio.adoc:
```diff
@@ -10,27 +10,44 @@ An installation of {ProductName} differs from upstream Istio community installat
+
+{ProductName} configures each member project to ensure network access between itself, the control plane, and other member projects. The exact configuration differs depending on how OpenShift software-defined networking (SDN) is configured. See xref:../../networking/openshift-sdn/about-openshift-sdn.adoc#about-openshift-sdn[About OpenShift SDN] for additional details.
+
+* *`NetworkPolicy`*: {ProductName} creates a `NetworkPolicy` resource in each member project allowing ingress to all pods from the other members and the control plane. If you remove a member from {ProductShortName}, this `NetworkPolicy` resource is deleted from the project.
```
These bullets appear have no lead-in sentence. Add something like "If the OCP cluster is configured to use the SDN plugin:"

geekspertise replied:
Fixed.

luksa commented on modules/ossm-vs-istio.adoc:
```diff
@@ -10,27 +10,44 @@ An installation of {ProductName} differs from upstream Istio community installat
 ====
 
-{ProductName} allows you to configure multi-tenant control plane installations, specify the namespaces that can access its {ProductShortName}, and isolate the {ProductShortName} from other control plane instances.
+* *Multitenant*: {ProjectName} joins the `NetNamespace` for each member project to the `NetNamespace` of the control plane project (for example, invoking oc `adm pod-network join-projects --to control-plane-project member-project`). If you remove a member from the {ProductShortName}, its `NetNamespace` is isolated from the control plane (for example, invoking `oc adm pod-network isolate-projects  ...
```
replace "for example, invoking" with "the equivalent of running"

geekspertise replied:
Fixed.

luksa commented on modules/ossm-vs-istio.adoc:
```diff
@@ -10,27 +10,44 @@ An installation of {ProductName} differs from upstream Istio community installat
 
 [id="ossm-rbac_{context}"]
 == Role Based Access Control features
 Role Based Access Control (RBAC) provides a mechanism you can use to control access to a service. You can identify subjects by username or by specifying a set of properties and apply access controls accordingly.
```
I believe this needs to be "Istio Role Based Access Control" to distinguish it from Kubernetes RBAC, which controls access to the API Server (whereas Istio RBAC controls access to services, as this sentence explains).

geekspertise replied:
Fixed.

luksa commented on modules/ossm-vs-istio.adoc:
```diff
@@ -136,4 +88,4 @@ All OpenShift routes created by {ProductName} are in the `istio-system` namespac
 
 [id="ossm-cni_{context}"]
 == Container Network Interface (CNI)
```
Add "Plugin" (and possibly "Istio" too - e.g. "Istio Container Network Interface (CNI) Plugin")

geekspertise replied:
Fixed.

luksa commented on modules/ossm-vs-istio.adoc:
```diff
@@ -136,4 +88,4 @@ All OpenShift routes created by {ProductName} are in the `istio-system` namespac
 == Container Network Interface (CNI)
-{ProductName} includes CNI which provides you with an alternate way to configure application pod networking. When you enable CNI, it replaces the `init-container` network configuration eliminating the need to grant service accounts and namespaces additional privileges by modifying their Security Context Constraints (SCCs). 
\ No newline at end of file
+{ProductName} includes CNI which provides you with an alternate way to configure application pod networking. CNI replaces the `init-container` network configuration eliminating the need to grant service accounts and projects additional privileges by modifying their Security Context Constraints (SCCs).
```
"includes _a_ CNI _plugin_..."

"_The_ CNI _plugin_ replaces..."

Replace:
"eliminating the need to grant service accounts and projects additional privileges by modifying their Security Context Constraints (SCCs)."

with:

"eliminating the need to grant service accounts and projects access to Security Context Constraints (SCCs) with elevated privileges"

geekspertise replied:
Fixed.

geekspertise commented on modules/ossm-cr-mixer.adoc:
```diff
@@ -1,28 +1,28 @@
-             memory: 4G
+    policy:
+        autoscaleEnabled: false
+
```
Fixed.

geekspertise commented on modules/ossm-cr-mixer.adoc:
```diff
@@ -1,28 +1,28 @@
-             cpu: 500m
-             memory: 4G
+    policy:
+        autoscaleEnabled: false
```
Fixed.

geekspertise commented on modules/ossm-cr-mixer.adoc:
```diff
@@ -1,28 +1,28 @@
+        autoscaleEnabled: false
+
+    telemetry:
+        autoscaleEnabled: false
```
Fixed.

geekspertise commented on modules/ossm-installation-activities.adoc:
```diff
@@ -5,17 +5,12 @@
 = {ProductName} installation activities
 
-The {ProductName} installation process creates two different projects (namespaces):
+You first deploy the {ProductName} Operator. This Operator defines and monitors the `ServiceMeshControlPlane` resource that manages the deployment, updating, and deletion of the {ProductShortName} components.
```
Fixed.

geekspertise commented on modules/ossm-installation-activities.adoc:
```diff
@@ -5,17 +5,12 @@
+* *{ProductName}* - based on the  open source link:https://istio.io/[Istio] project, lets you connect, secure, control, and observe the microservices that make up your applications.
 
-You first create a Kubernetes _Operator_. This Operator defines and monitors a _custom resource_ that manages the deployment, updating, and deletion of the {ProductShortName} components.
+Depending on how you define the custom resource, you can install one or more of the following components when you install the {ProductShortName}:
```
This is no longer valid. This section was reworked after a review with Support.

geekspertise commented on modules/ossm-cr-istio-global.adoc:
```diff
@@ -16,8 +16,8 @@ In order for the 3scale Istio Adapter to work, `disablePolicyChecks` must be `fa
-      hub: `maistra/` or `registry.redhat.io/openshift-istio-tech-preview/`
-      tag: 0.12.0
+      tag: 1.0.0
+      hub: maistra/ or registry.redhat.io/openshift-service-mesh/
```
I removed "maistra/" to alleviate confusion. The reader can see all of the available parameters below.

geekspertise commented on modules/ossm-member-roll.adoc:
```diff
@@ -0,0 +1,89 @@
+The `ServiceMeshMemberRoll` is deleted when its corresponding `ServiceMeshControlPlane` resource is deleted.
+
+[id="ossm-member-roll-console_{context}"]
+== Configure the member roll from the the {product-title} web console
```
Fixed.

geekspertise commented on modules/ossm-cr-example.adoc:
```diff
@@ -1,16 +1,20 @@
-To deploy the {ProductShortName} control plane, you must create a custom resource. A _custom resource_ allows you to introduce your own API into a Kubernetes project or cluster. You create a custom resource yaml file that defines the project parameters and creates the object. This example custom resource yaml file contains all of the supported parameters and deploys {ProductName} {ProductVersion} ...
+A _custom resource_ allows you to extend the API in an {ProductName} project or cluster. When you deploy {ProductShortName} it creates a default `ServiceMeshControlPlane` that you can modify to change the project parameters.
+
+The {ProductShortName} operator extends the API by adding the `ServiceMeshControlPlane` resource, which enables you to create `ServiceMeshControlPlane` objects within projects. By creating a `ServiceMeshControlPlane` object, you instruct the Operator to install a {ProductShortName} control plane into the project, configured with the parameters you set in the `ServiceMeshControlPlane` object.
```
Fixed.

geekspertise commented on modules/ossm-member-roll.adoc:
```diff
@@ -0,0 +1,89 @@
+
+. Log into the {product-title} CLI. 
+
+. Create `ServiceMeshMemberRoll` object in the same project as the `ServiceMeshControlPlane` resource.
```
Changed it to resource.

geekspertise commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,168 @@
+[id="ossm-operatorhub-install_{context}"]
+= Installing the Operators from OperatorHub
+
+The {ProductShortName} installation process uses the link:https://operatorhub.io/[OperatorHub] to manage the installation of the control plane within the `openshift-operators` project. The OperatorHub defines and monitors the `ServiceMeshControlPlane` related to the deployment, update, and deletion of the control plane.
```
Updated to read:

"The {ProductShortName} installation process uses the link:https://operatorhub.io/[OperatorHub] to install the `ServiceMeshControlPlane` custom resource definition within the `openshift-operators` project. The {ProductName} defines and monitors the `ServiceMeshControlPlane` related to the deployment, update, and deletion of the control plane."

geekspertise commented on modules/ossm-control-plane-templates.adoc:
```diff
@@ -0,0 +1,82 @@
+$ oc create configmap --from-file=<templates-directory> smcp-templates -n openshift-operators
+----
+
+. Locate the Operator deployment.
```
Fixed.

geekspertise commented on modules/ossm-control-plane-templates.adoc:
```diff
@@ -0,0 +1,82 @@
+
+You can create reusable configurations with templates. Individual users can extend the templates you create with their own configurations. Templates can also inherit configuration information from other templates. For example, you can create an accounting control plane for the accounting team and a marketing control plane for the marketing team. If you create a development template and a producti ...
+
+When you configure control plane templates, which follow the same syntax as the `ServiceMeshControlPlane`, users inherit settings in a hierarchical fashion. The Operator is delivered with a `default` template with default settings for {ProductName}. To add custom templates you must create a ConfigMap named `smcp-templates` in the `openshift-operators` and mount the ConfigMap in the Operator at `/ ...
```
Fixed.

geekspertise commented on modules/ossm-control-plane-templates.adoc:
```diff
@@ -0,0 +1,82 @@
+[id="ossm-control-plane-templates_{context}"]
+= Creating control plane templates
+
+You can create reusable configurations with templates. Individual users can extend the templates you create with their own configurations. Templates can also inherit configuration information from other templates. For example, you can create an accounting control plane for the accounting team and a marketing control plane for the marketing team. If you create a development template and a producti ...
```
Fixed.

geekspertise commented on modules/ossm-control-plane-remove.adoc:
```diff
@@ -0,0 +1,69 @@
+
+. Log into the {product-title} CLI.
+
+. Run this command to retrieve the name of the installed custom resource:
```
Fixed.

geekspertise commented on modules/ossm-control-plane-remove.adoc:
```diff
@@ -0,0 +1,69 @@
+
+[TIP]
+====
+You can use the shortened `smcp` command in place of `servicemeshcontrolplane`.
```
Fixed.

geekspertise commented on modules/ossm-control-plane-remove.adoc:
```diff
@@ -0,0 +1,69 @@
+
+[id="ossm-control-plane-remove_{context}"]
+= Removing the {ProductName} control plane
+You can remove the {ProductShortName} control plane by using the {product-title} web console or the CLI.
```
When I built locally it rendered.

geekspertise commented on modules/ossm-control-plane-deploy.adoc:
```diff
@@ -3,37 +3,74 @@
+
+. Under *Provided APIs*, the Operator enables you to create two resource types:
+** A `ServiceMeshControlPlane` resource
+** A `ServiceMeshMemberRoll`
```
Fixed.

geekspertise commented on modules/ossm-control-plane-deploy.adoc:
```diff
@@ -3,37 +3,74 @@
 
 [id="ossm-control-plane-deploy_{context}"]
+= Deploying the {ProductName} control plane
+You can deploy the {ProductShortName} control plane by using the {product-title} web console or the CLI.
```
Fixed.

geekspertise commented on modules/ossm-configuring-kiali.adoc:
```diff
@@ -0,0 +1,98 @@
+* Jaeger authorization for the Kiali console
+* Jaeger URL for the Kiali console
+
+Kiali can automatically detect the Jaeger URL. However if you have a custom Jaeger installation that is not easily auto-detectable by Kiali, you must update the URL value in the `ServiceMeshControlPlane` file.
```
Changed it to resource.

geekspertise commented on modules/ossm-configuring-kiali.adoc:
```diff
@@ -0,0 +1,98 @@
+* Grafana authorization for the Kiali console
+* Grafana URL for the Kiali console
+
+Kiali can automatically detect the Grafana URL. However if you have a custom Grafana installation that is not easily auto-detectable by Kiali, you must update the URL value in the `ServiceMeshControlPlane` file.
```
Changed this to resource.

geekspertise commented on modules/ossm-configuring-kiali.adoc:
```diff
@@ -0,0 +1,98 @@
+
+|ingress
+   enabled
+|This enables/disables ingress.
```
@jstickler can address this Post-GA.

geekspertise commented on modules/ossm-configuring-kiali.adoc:
```diff
@@ -0,0 +1,98 @@
+    enabled: true
+    dashboard:
+      viewOnlyMode: false
+      jaegerURL: "http://jaeger-query-istio-system.127.0.0.1.nip.io"
```
@JStickler can address this Post-GA since the parameters were not provided for the table.

geekspertise commented on modules/ossm-configuring-kiali.adoc:
```diff
@@ -0,0 +1,98 @@
+    enabled: true
+    dashboard:
+      viewOnlyMode: false
+      grafanaURL:  "https://grafana-istio-system.127.0.0.1.nip.io"
```
@JStickler can address this Post-GA since the parameters were not provided for the table.

geekspertise commented on modules/ossm-configuring-kiali.adoc:
```diff
@@ -0,0 +1,98 @@
+[id="configuring-kiali_{context}"]
+= Configuring Kiali
+
+When the {ProductShortName} Operator creates the `ServiceMeshControlPlane` it also creates the custom resource definition (CRD) for Kiali. The Kiali operator then uses this CRD when creating Kaili instances.
```
Updated to read:

"When the {ProductShortName} Operator creates the `ServiceMeshControlPlane` it also processes the Kiali resource. The Kiali Operator then uses this object when creating Kaili instances."

geekspertise commented on modules/ossm-configuring-jaeger.adoc:
```diff
@@ -0,0 +1,131 @@
+For this release, the only supported storage option for production use is Elasticsearch.
+====
+
+Elasticsearch is a memory intensive application. The initial set of nodes created by the installation may not be large enough to support the Elasticsearch cluster. Additional nodes must be added to the  cluster if you want to run with the recommended (or better) memory. Each ES node can operate with a lower memory setting though this is not recommended for production deployments.
```
Updated this to read:

"Elasticsearch is a memory intensive application. The initial set of nodes created by the {product-title} installation may not be large enough to support the Elasticsearch cluster. Additional nodes must be added to the  cluster if you want to run with the recommended amount (or more) memory. Each ES node can operate with a lower memory setting though this is not recommended for production deployments."

geekspertise commented on modules/ossm-configuring-jaeger.adoc:
```diff
@@ -0,0 +1,131 @@
+|`true`/`false`
+|`true`
+
+|ingress
```
This is unclear. @JStickler can address this post-GA.

geekspertise commented on modules/ossm-configuring-jaeger.adoc:
```diff
@@ -0,0 +1,131 @@
+[id="ossm-configuring-jaeger_{context}"]
+= Configuring Jaeger
+
+When the {ProductShortName} Operator creates the `ServiceMeshControlPlane` it also creates the custom resource definition (CRD) for Jaeger. The Jaeger operator then uses this CRD when creating Jaeger instances.
```
Updated this to read:

"When the {ProductShortName} Operator creates the `ServiceMeshControlPlane` resource it also creates the Jaeger resource. The Jaeger Operator then uses this object when creating Jaeger instances."

geekspertise commented on modules/ossm-automatic-sidecar-injection.adoc:
```diff
@@ -4,7 +4,7 @@
 [id="ossm-automatic-sidecar-injection_{context}"]
 = Enabling automatic sidecar injection
-When deploying an application into the {ProductName} you must opt in to injection by specifying the `sidecar.istio.io/inject` annotation with a value of `true`. Opting in ensures that the sidecar injection does not interfere with other OpenShift features such as builder pods used by numerous frameworks within the OpenShift ecosystem.
+When deploying an application into the {ProductName} you must opt in to injection by specifying the `sidecar.istio.io/inject` annotation with a value of `"true"`. Opting in ensures that the sidecar injection does not interfere with other OpenShift features such as builder pods used by numerous frameworks within the OpenShift ecosystem.
```
Fixed.

geekspertise replied:
Fixed.

geekspertise commented on modules/ossm-configuring-jaeger.adoc:
```diff
@@ -0,0 +1,131 @@
+        jaeger:
+          elasticsearch:
+            nodeCount: 3
+            redundancyPolicy:
```
@JStickler can address this post-GA.

geekspertise commented:
@sheriff-rh I finished my (known) updates to this GA PR for the Service Mesh Install and Release Notes documentation for GA. I squashed and rebased and fixed the topic yaml (for now -- it needs to be adjusted when we're ready to go live). Can you start your review? Thanks!

ahardin-rh commented on service_mesh/service_mesh_day_two/ossm-example-bookinfo.adoc:
```diff
@@ -5,7 +5,12 @@ include::modules/ossm-document-attributes.adoc[]
-// The following include statements pull in the module files that comprise the assembly.
+[WARNING]
+====
+The Bookinfo example application allows you to test your {ProductName} {ProductVersion} install on {product-title} 4.1.
```
s/install/installation

geekspertise replied:
Fixed.

ahardin-rh commented on modules/ossm-vs-istio.adoc:
```diff
@@ -65,75 +82,12 @@ spec:
-== Container Network Interface (CNI)
-{ProductName} includes CNI which provides you with an alternate way to configure application pod networking. When you enable CNI, it replaces the `init-container` network configuration eliminating the need to grant service accounts and namespaces additional privileges by modifying their Security Context Constraints (SCCs). 
\ No newline at end of file
+== The Istio Container Network Interface (CNI) plugin
```
s/plugin/plug-in

geekspertise replied:
Fixed.

ahardin-rh commented on modules/ossm-vs-istio.adoc:
```diff
@@ -10,27 +10,46 @@ An installation of {ProductName} differs from upstream Istio community installat
-== Role Based Access Control features
-Role Based Access Control (RBAC) provides a mechanism you can use to control access to a service. You can identify subjects by username or by specifying a set of properties and apply access controls accordingly.
+== Istio Role Based Access Control features
+Istio Role Based Access Control (RBAC) provides a mechanism you can use to control access to a service. You can identify subjects by username or by specifying a set of properties and apply access controls accordingly.
```
s/username/user name

geekspertise replied:
Fixed.

ahardin-rh commented on modules/ossm-vs-istio.adoc:
```diff
@@ -10,27 +10,46 @@ An installation of {ProductName} differs from upstream Istio community installat
+
+{ProductName} configures each member project to ensure network access between itself, the control plane, and other member projects. The exact configuration differs depending on how OpenShift software-defined networking (SDN) is configured. See xref:../../networking/openshift-sdn/about-openshift-sdn.adoc#about-openshift-sdn[About OpenShift SDN] for additional details.
+
+If the {product-title} cluster is configured to use the SDN plugin:
```
s/plugin/plug-in

geekspertise replied:
Fixed.

ahardin-rh commented on modules/ossm-tutorial-bookinfo-removing.adoc:
```diff
@@ -6,37 +6,55 @@ This TASK module included in the following assemblies:
 
+.Procedure
+
+. Log into the {product-title} web console.
```
s/Log into/Log in to

geekspertise replied:
Fixed.

ahardin-rh commented on modules/ossm-tutorial-bookinfo-overview.adoc:
```diff
@@ -6,9 +6,9 @@ This CONCEPT module included in the following assemblies:
 = Bookinfo application
 
-The upstream Istio project has an example tutorial called https://istio.io/docs/examples/bookinfo[bookinfo], which is composed of four separate microservices used to demonstrate various Istio features.  The Bookinfo application displays information about a book, similar to a single catalog entry of an online book store. Displayed on the page is a description of the book, book details (ISBN, numbe ...
+The upstream Istio project has an example tutorial called https://istio.io/docs/examples/bookinfo[bookinfo], which is composed of four separate microservices used to demonstrate various Istio features. The Bookinfo application displays information about a book, similar to a single catalog entry of an online book store. Displayed on the page is a description of the book, book details (ISBN, number ...
```
pages, and other information

geekspertise replied:
Fixed.

ahardin-rh commented on modules/ossm-tutorial-bookinfo-install.adoc:
```diff
@@ -6,55 +6,85 @@ This TASK module included in the following assemblies:
+
+. Click *Create Service Mesh Member Roll*.
+
+. Edit the default Service Mesh Member Roll yaml and add `bookinfo` to the *members* list.
```
s/yaml/YAML file

geekspertise replied:
Fixed.

ahardin-rh commented on modules/ossm-tutorial-bookinfo-install.adoc:
```diff
@@ -6,55 +6,85 @@ This TASK module included in the following assemblies:
 
-. Create a project for the Bookinfo application.
-+
+. Log into the {product-title} web console.
```
s/log into/log in to

geekspertise replied:
Fixed.

ahardin-rh commented on modules/ossm-tutorial-bookinfo-adding-destination-rules.adoc:
```diff
@@ -6,29 +6,20 @@ This TASK module included in the following assemblies:
 = Adding default destination rules
 
-Before you can use the Bookinfo application, you have to add default destination rules.  There are two preconfigured yaml files, depending on whether or not you enabled mutual transport layer security (TLS) authentication.
+Before you can use the Bookinfo application, you have to add default destination rules. There are two preconfigured yaml files, depending on whether or not you enabled mutual transport layer security (TLS) authentication.
```
s/yaml/YAML

geekspertise replied:
Fixed.

ahardin-rh commented on modules/ossm-rn-new-features.adoc:
```diff
@@ -11,6 +11,19 @@ Reason – If known, include why has the enhancement been implemented (use case,
+
+Other notable changes in this release include the following:
+
+* The Kubernetes Container Network Interface (CNI) plugin is always on.
```
s/plugin/plug-in

geekspertise replied:
Fixed.

ahardin-rh commented on modules/ossm-rn-fixed-issues.adoc:
```diff
@@ -16,62 +16,30 @@ Result - How has the behavior changed as a result?  Try to avoid “It is fixed
+* link:https://issues.jboss.org/browse/MAISTRA-684[MAISTRA-684] The default Jaeger version in the `istio-operator` is 1.12.0, which does not match Jaeger version 1.13.1 that shipped in {ProductName} 0.12.TechPreview.
 
-* https://issues.jboss.org/browse/MAISTRA-13[MAISTRA-13] - Installer should determine release version from installed components.  At present, the installer queries the yaml file, however if the yaml has been modified, the installer is not able to remove an older version.
+* link:https://issues.jboss.org/browse/MAISTRA-572[MAISTRA-572] Jaeger cannot be used with Kiali. In this release Jaeger is configured to use the OAuth proxy, but is also only configured to work through a browser and doesn't allow service access. Kiali cannot properly communicate with the Jaeger endpoint and it considers Jaeger to be disabled. See also link:https://issues.jboss.org/browse/TRACING ...
```
s/doesn't/does not

geekspertise replied:
Fixed.

ahardin-rh commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,121 @@
+
+.Procedure
+
+. Log into the {product-title} web console.
```
s/Log into/Log in to

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-mixer-policy.adoc:
```diff
@@ -8,9 +8,14 @@
 
 .Procedure
 
+. Log into the {product-title} CLI. 
```
IBM:

> Log into v
> Do not use. Use “log in to.”

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,168 @@
+
+.Procedure
+
+. Log into the {product-title} web console.
```
IBM:

> Log into v
> Do not use. Use “log in to.”

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,168 @@
+
+. Click *Install*.
+
+. On the *Create Operator Subscription* page, select *All namespaces on the cluster (default)*. This installs the Operator in the default `openshift-operators` project and is made available to all projects in the cluster.
```
What is made available? The Operator I presume?

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,168 @@
+
+. On the *Create Operator Subscription* page, select *All namespaces on the cluster (default)*. This installs the Operator in the default `openshift-operators` project and is made available to all projects in the cluster.
+
+. Select the *preview* Update Channel.
```
Are we capitalizing _Update Channel_? I'm not familiar with it.

geekspertise replied:
"Update Channel" is capitalized in the UI.

jboxman commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,168 @@
+
+. Click *Subscribe*.
+
+. The *Subscription Overview* page displays the Elasticsearch Operator's install process.
```
_install process_ or _installation progress_?

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,168 @@
+
+.Procedure
+
+. Log into the {product-title} web console.
```
_Log in to_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,168 @@
+
+. Click *Install*.
+
+. On the *Create Operator Subscription* page, select *All namespaces on the cluster (default)*. This installs the Operator in the default `openshift-operators` project and is made available to all projects in the cluster.
```
Possibly: "This installs the Operator in the default `openshift-operators` project and _makes the Operator_ available to all projects in the cluster."

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,168 @@
+
+. Click *Subscribe*.
+
+. The *Subscription Overview* page displays the Jaeger Operator's install process.
```
_installation progress_?

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,168 @@
+
+.Procedure
+
+. Log into the {product-title} web console.
```
_Log in to_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,168 @@
+
+. Click *Install*.
+
+. On the *Create Operator Subscription* page, select *All namespaces on the cluster (default)*. This installs the Operator in the defaut `openshift-operators` project and is made available to all projects in the cluster.
```
Same comment as for line 77.

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,168 @@
+
+. Click *Subscribe*.
+
+. The *Subscription Overview* page displays the Kiali Operator's install process.
```
Same comment as line 90.

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,168 @@
+
+.Procedure
+
+. Log into the {product-title} web console.
```
_Log in to_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,168 @@
+
+. Click *Subscribe*.
+
+. The *Subscription Overview* page displays the {ProductName} Operator's install process.
```
Same comment as line 90.

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,121 @@
+
+.Procedure
+
+. Log into the {product-title} web console.
```
_Log in to_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,121 @@
+
+. Click the *servicemeshoperator* Options menu {kebab}, and then click *Remove Subscription*.
+
+. Select *Also completely remove the Operator from the selected namespace* and click *Remove*.
```
_namespace, and then click_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,121 @@
+
+.Procedure
+
+. Log into the {product-title} web console.
```
_Log in to_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,121 @@
+
+. Click the *elasticsearch-operator* Options menu {kebab}, and then click *Remove Subscription*.
+
+. Select *Also completely remove the Operator from the selected namespace* and click *Remove*.
```
_, and then click_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,121 @@
+
+.Procedure
+
+. Log into the {product-title} web console.
```
_Log in to_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,121 @@
+
+. Click the *jaegeroperator* Options menu {kebab}, and then click *Remove Subscription*.
+
+. Select *Also completely remove the Operator from the selected namespace* and click *Remove*.
```
_, and then click_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,121 @@
+
+.Procedure
+
+. Log into the {product-title} web console.
```
_Log in to_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,121 @@
+
+. Click the *kiali-ossm* Options menu {kebab}, and then click *Remove Subscription*.
+
+. Select *Also completely remove the Operator from the selected namespace* and click *Remove*.
```
_, and then click_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,121 @@
+
+.Procedure
+
+. Log into the {product-title} CLI as a cluster administrator.
```
_Log in to_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,121 @@
+$ oc delete validatingwebhookconfiguration/<operator-project>.servicemesh-resources.maistra.io
+$ oc delete -n <operator-project> daemonset/istio-node
+$ oc delete clusterrole/istio-admin
+$ oc get crds -o name | grep '.*\.istio\.io' |xargs -r -n 1 oc delete
```
For consistency I'd include a space between `|` and `xargs`, but that's stylistic and `bash` won't care.

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-rn-fixed-issues.adoc:
```diff
@@ -16,62 +16,30 @@ Result - How has the behavior changed as a result?  Try to avoid “It is fixed
-* https://issues.jboss.org/browse/MAISTRA-462[MAISTRA-462] [Multi-tenant implementation] After adding a namespace member to a second control plane, Kiali does not display the namespace member in the namespace list because the namespace for the second control plane is missing the `maistra.io/member-of` and `istio.openshift.io/member-of` labels. This is a result of the installation of the second co ...
-
-* https://issues.jboss.org/browse/MAISTRA-466[MAISTRA-466] [Multi-tenant implementation] When you install more than one control plane, the operator overwrites the Kiali oauthclient yaml on any existing Kiali installations so that only the most recent installation functions. The Kiali oauthclient (`oc get oauthclients kiali`) contains a list of the authorized URLs that OpenShift OAuth will redirec ...
+* link:https://issues.jboss.org/browse/MAISTRA-357[MAISTRA-357] In OpenShift 4 Beta on AWS, it is not possible, out of the box, to access a TCP or HTTPS service through the ingress gateway on a port other than port 80. The AWS load balancer has a health check that verifies if port 80 on the service endpoint is active.
```
_out of the box_ may not translate well; Perhaps _by default_?

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-rn-new-features.adoc:
```diff
@@ -11,6 +11,19 @@ Reason – If known, include why has the enhancement been implemented (use case,
+
+Other notable changes in this release include the following:
+
+* The Kubernetes Container Network Interface (CNI) plugin is always on.
```
_plug-in_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-threescale-integrate.adoc:
```diff
@@ -17,7 +17,7 @@ You can use these examples to configure requests to your services using the 3sca
 [NOTE]
 ====
-To configure the 3scale Istio Adapter, refer to xref:../service_mesh_install/installing-ossm.html#ossm-cr-parameters_installing-ossm[{ProductName} custom resources] for instructions on adding adapter parameters to the custom resource file.
+To configure the 3scale Istio Adapter, refer to xref:../service_mesh_install/customizing-installation-ossm.adoc#ossm-cr-parameters_customizing-installation-ossm[{ProductName} custom resources] for instructions on adding adapter parameters to the custom resource file.
```
I don't think we can `xref` in modules, though.

jboxman commented on modules/ossm-threescale-manifests.adoc:
```diff
@@ -23,7 +23,7 @@ oc exec -n ${NS} $(oc get po -n ${NS} -o jsonpath='{.items[?(@.metadata.labels.a
 
 
-. xref:../threescale_adapter/threescale-adapter.html#ossm-threescale-routing_threescale-adapter[Update the workload] with the required annotations:
+. xref:../threescale_adapter/threescale-adapter.adoc#ossm-threescale-routing_threescale-adapter[Update the workload] with the required annotations:
```
We haven't been using `xref` in modules:

https://redhat-documentation.github.io/modular-docs/#modular-docs-terms-definitions

geekspertise replied:
@jboxman I reverted this change, as this part of the documentation is handled by the 3scale team. @dfennessy please make a note of this for your documentation.

jboxman commented on modules/ossm-tutorial-bookinfo-adding-destination-rules.adoc:
```diff
@@ -6,29 +6,20 @@ This TASK module included in the following assemblies:
 = Adding default destination rules
 
-Before you can use the Bookinfo application, you have to add default destination rules.  There are two preconfigured yaml files, depending on whether or not you enabled mutual transport layer security (TLS) authentication.
+Before you can use the Bookinfo application, you have to add default destination rules. There are two preconfigured yaml files, depending on whether or not you enabled mutual transport layer security (TLS) authentication.
```
Uppercase _YAML_.

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-tutorial-bookinfo-install.adoc:
```diff
@@ -6,55 +6,85 @@ This TASK module included in the following assemblies:
 
-. Create a project for the Bookinfo application.
-+
+. Log into the {product-title} web console.
```
_Log in to_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-tutorial-bookinfo-install.adoc:
```diff
@@ -6,55 +6,85 @@ This TASK module included in the following assemblies:
+
+. Click *Create Service Mesh Member Roll*.
+
+. Edit the default Service Mesh Member Roll yaml and add `bookinfo` to the *members* list.
```
_YAML_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-tutorial-bookinfo-install.adoc:
```diff
@@ -6,55 +6,85 @@ This TASK module included in the following assemblies:
-
 +
-. Deploy the Bookinfo application in the _"myproject"_ Namespace by applying the `bookinfo.yaml` file:
+** Alternatively, you can run this command from the CLI to add the `bookinfo` project to the `ServiceMeshMemberRoll`. Replace `<control plane project>` with the name of your control plane project.
```
Use underscores for token:

```
<control_plane_product>
```

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-tutorial-bookinfo-install.adoc:
```diff
@@ -6,55 +6,85 @@ This TASK module included in the following assemblies:
 
+. Click *Create* to save the updated Service Mesh Member Roll.
+
+. From the CLI, deploy the Bookinfo application in the _"bookinfo"_ project by applying the `bookinfo.yaml` file:
```
`bookinfo` with back ticks.

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-tutorial-bookinfo-overview.adoc:
```diff
@@ -6,9 +6,9 @@ This CONCEPT module included in the following assemblies:
 = Bookinfo application
 
-The upstream Istio project has an example tutorial called https://istio.io/docs/examples/bookinfo[bookinfo], which is composed of four separate microservices used to demonstrate various Istio features.  The Bookinfo application displays information about a book, similar to a single catalog entry of an online book store. Displayed on the page is a description of the book, book details (ISBN, numbe ...
+The upstream Istio project has an example tutorial called https://istio.io/docs/examples/bookinfo[bookinfo], which is composed of four separate microservices used to demonstrate various Istio features. The Bookinfo application displays information about a book, similar to a single catalog entry of an online book store. Displayed on the page is a description of the book, book details (ISBN, number ...
```
"ISBN, number of pages, and other information" - comma after pages.

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-tutorial-bookinfo-removing.adoc:
```diff
@@ -6,37 +6,55 @@ This TASK module included in the following assemblies:
 
-. Download the cleanup script:
-+
+. Log into the {product-title} web console.
```
_Log in to_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-tutorial-bookinfo-removing.adoc:
```diff
@@ -6,37 +6,55 @@ This TASK module included in the following assemblies:
-+
+. Click on the `bookinfo` menu {kebab} and choose *Delete Project*.
 
+. Type `bookinfo` in the confirmation dialog box and click *Delete*.
```
_, and then_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-tutorial-bookinfo-removing.adoc:
```diff
@@ -6,37 +6,55 @@ This TASK module included in the following assemblies:
 
-. Delete the Bookinfo virtualservice, gateway, and terminate the Pods by running the cleanup script:
-+
+. Click on the `bookinfo` menu {kebab} and choose *Delete Project*.
```
Possibly: Click on the `bookinfo` menu {kebab}, and then click *Delete Project*.

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-tutorial-bookinfo-removing.adoc:
```diff
@@ -6,37 +6,55 @@ This TASK module included in the following assemblies:
 
+.Procedure
+
+. Log into the {product-title} web console.
```
_Log in to_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-tutorial-bookinfo-verify-install.adoc:
```diff
@@ -8,19 +8,26 @@ This TASK module included in the following assemblies:
 .Procedure
 
-* To confirm that the application is deployed:
+. Log into the {product-title} CLI. 
```
_Log in to_

geekspertise replied:
Fixed.

jboxman commented on modules/ossm-tutorial-bookinfo-removing.adoc:
```diff
@@ -6,37 +6,55 @@ This TASK module included in the following assemblies:
+
+. Click the `ServiceMeshMemberRoll` menu {kebab} and select *Edit Service Mesh Member Roll*.
+
+. Edit the default Service Mesh Member Roll yaml and remove `bookinfo` from the *members* list.
```
YAML

geekspertise replied:
Fixed.

knrc commented on modules/ossm-member-roll.adoc:
```diff
@@ -0,0 +1,89 @@
+
+. Click the *Istio Service Mesh Member Roll* link under *Provided APIs* for the *{ProductName}* Operator.
+
+. Click *Create Service Mesh Member Roll*.
```
Note the "404" @travisrogers05 has been seeing seems to be an issue with OLM and how it configures the namespaces, it may be necessary to refresh the console before you are presented with the *Create Istio Service Mesh Member Roll* button.

geekspertise replied:
@travisrogers05 and @knrc I added the following note:

[NOTE]
====
It can take a short time for the Operator to finish creating the projects, therefore you may need to refresh the screen before the web console presents the *Create Istio Service Mesh Member Roll* button.
====

knrc commented on modules/ossm-configuring-jaeger.adoc:
```diff
@@ -0,0 +1,131 @@
+Elasticsearch is a memory intensive application. The initial set of nodes created by the {product-title} installation may not be large enough to support the Elasticsearch cluster. Additional nodes must be added to the  cluster if you want to run with the recommended amount (or more) memory. Each ES node can operate with a lower memory setting though this is not recommended for production deployme ...
+
+You should modify the default Elasticsearch configuration to match your use case. You can adjust both the CPU and memory limits for each component by modifying the resources block with valid memory and CPU values.
+
```
The elastic search default resources now prevent Service Mesh from installing cleanly on the default OpenShift installation, we will need to include some text saying that the production-elasticsearch should not be used unless the user has created a large cluster with significant resources sufficient to support this installation.  Our preference is likely now to be favouring the all-in-one as default for Service Mesh installations with the proviso customers should manually update to production-elasticsearch before running in a production setting.

knrc commented on modules/ossm-cr-example.adoc:
```diff
@@ -67,9 +67,10 @@ The 3scale Istio Adapter is deployed and configured in the custom resource file.
       tracing:
         enabled: true
+        jaeger:
+          template: all-in-one
```
When installing through the openshift console we should also point out that customers should switch the template from the default, production-elasticsearch, to the above all-in-one variant for all but large scale production use cases.

knrc replied:
The above will no longer be necessary, we've decided to update the metadata and republish the operator image using all-in-one as the defgault

vikram-redhat commented:
@geekspertise the peer review is done.

geekspertise commented on modules/ossm-operatorhub-install.adoc:
```diff
@@ -0,0 +1,168 @@
+. Click the {ProductName} Operator to display information about the Operator.
+
+. Click *Install*.
+
```
Fixed.

geekspertise commented on modules/ossm-member-roll.adoc:
```diff
@@ -0,0 +1,89 @@
+// * service_mesh/service_mesh_install/installing-ossm.adoc
+
+[id="ossm-member-roll_{context}"]
+= Configure the {ProductName} member roll
```
I've moved this module to the end of the installing-ossm.adoc assembly. It appears after ossm-control-plane-deploy.adoc now.

geekspertise commented on modules/ossm-member-roll.adoc:
```diff
@@ -0,0 +1,89 @@
+
+. Click *Catalog* -> *Installed Operators*.
+
+. Click the *Project* menu and choose `openshift-operators` from the list.
```
I changed this, to istio-system, but if I need to change it back, let me know @travisrogers05.

geekspertise commented on modules/ossm-control-plane-remove.adoc:
```diff
@@ -0,0 +1,70 @@
+
+. Navigate to *Catalogs* -> *Installed Operators*.
+
+. Click the *Project* menu and choose a project from the list.
```
Fixed.

geekspertise commented on modules/ossm-control-plane-remove.adoc:
```diff
@@ -0,0 +1,70 @@
+
+. Log into the {product-title} web console.
+
+. Navigate to *Catalogs* -> *Installed Operators*.
```
Fixed.

geekspertise commented on modules/ossm-control-plane-remove.adoc:
```diff
@@ -0,0 +1,70 @@
+
+. Click the *Project* menu and choose a project from the list.
+
+. Click the `ServiceMeshControlPlane` menu {kebab}.
```
Fixed.

geekspertise commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,121 @@
+. Select *Also completely remove the Operator from the selected namespace* and click *Remove*.
+
+[id="ossm-remove-operator-elasticsearch_{context}"]
+== Removing the Elasticsearch Operator
```
I updated this and updated the intro as well to reflect the removal order change.

geekspertise commented on modules/ossm-operatorhub-remove.adoc:
```diff
@@ -0,0 +1,121 @@
+. Select *Also completely remove the Operator from the selected namespace* and click *Remove*.
+
+[id="ossm-remove-operator-kiali_{context}"]
+== Removing the Jaeger Operator
```
Fixed.

knrc commented on modules/ossm-control-plane-deploy.adoc:
```diff
@@ -3,37 +3,75 @@
+
+. Click *Create New* under *Istio Service Mesh Control Plane*.
+
+. Modify the minimal `ServiceMeshControlPlane` template.
```
Note the default within the metadata and operator image is now to use all-in-one instead of production-elasticsearch.

knrc commented on modules/ossm-configuring-jaeger.adoc:
```diff
@@ -0,0 +1,131 @@
+
+[NOTE]
+====
+For this release, the only supported storage option for production use is Elasticsearch.
```
@geekspertise @neal-timpe 

For this release, the default template strategy used within the `ServiceMeshControlPlane` resource is `AllInOne`.  For production use the only supported storage option is Elasticsearch therefore the `ServiceMeshControlPlane` must be configured to request the `production-elasticsearch` template strategy when deploying Service Mesh within a production environment.

knrc commented on modules/ossm-configuring-jaeger.adoc:
```diff
@@ -0,0 +1,131 @@
+You should modify the default Elasticsearch configuration to match your use case. You can adjust both the CPU and memory limits for each component by modifying the resources block with valid memory and CPU values.
+
+
+.Default Jaeger parameters
```
Default Elasticsearch parameters when using the `production-elasticsearch` jaeger template.

knrc commented on modules/ossm-configuring-jaeger.adoc:
```diff
@@ -0,0 +1,131 @@
+  spec:
+    istio:
+      tracing:
+        jaeger:
```
Add in template: production-elasticsearch at the same indentation as the following elasticsearch element.

vikram-redhat commented:
Got confirmation that this is now ok to merge and cherry pick.

vikram-redhat commented:
/cherrypick enterprise-4.1

vikram-redhat commented:
/cherrypick enterprise-4.2
